### PR TITLE
[dashboard] Show last data load time

### LINKED ISSUE #49011: [Dashboard] Provide an ability to know the staleness of the data shown in the UI
### Description

It can be hard to know if the dashboard view is stale or not. It would be good to know the last load time of data and perhaps a button to refresh the data manually.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by tdat1465**
## Why are these changes needed?

Fixes #49011.

The Ray Dashboard can show data that may be stale without giving users a clear indication of when the data was last loaded. This change adds a small UI affordance in the main dashboard navigation to show the latest successful dashboard data load time and provide a manual refresh action.

## What changed?

- Tracks successful dashboard API responses through the shared request handler.
- Adds a localized freshness component in the top navigation so only the freshness UI re-renders when dashboard data loads.
- Shows a freshness label such as `Updated just now` or `Updated 2m ago`.
- Adds a tooltip with the exact last successful data load time, respecting the dashboard's selected timezone.
- Adds a manual refresh button that revalidates mounted SWR data.
- Adds test coverage for freshness label formatting.

## Team contributions

- @honorPND - Pham Ngoc Duy - 23120035: helped investigate the dashboard staleness issue and reviewed the UI behavior.
- @lhmha - Le Hoang My Ha - 23120038: helped test the dashboard frontend/backend locally and verified the freshness indicator behavior.
- @ohthisischichi - Phan Thi Phuong Chi - 23120025: helped review the implementation and PR description.
- @tdat1465 - Tran Thanh Dat - 23120030: implemented the dashboard freshness indicator, manual refresh action, and related tests.

## How was this tested?

- `npm.cmd exec -- prettier --write src/App.tsx src/common/constants.ts src/service/requestHandlers.ts src/pages/layout/MainNavLayout.tsx src/pages/layout/MainNavLayout.component.test.tsx`
- `npm.cmd exec -- eslint src/App.tsx src/common/constants.ts src/service/requestHandlers.ts src/pages/layout/MainNavLayout.tsx src/pages/layout/MainNavLayout.component.test.tsx`
- `npm.cmd exec -- tsc --noEmit`
- `$env:CI='true'; npm.cmd --prefix python\ray\dashboard\client test -- --watchAll=false --runInBand src/pages/layout/MainNavLayout.component.test.tsx`
- Manually verified with the dashboard frontend that the top nav shows `Updated just now`, updates over time, shows the exact load timestamp on hover, respects the selected timezone, and refreshes data through the refresh button.

## Checks

- [x] I've signed off every commit (by using the `-s` flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/.

## Testing Strategy

- [x] Unit tests
- [x] Manual tests
- [ ] This PR is not tested :(

Screenshot 2026-05-25 145317

### CODE REVIEW

File: python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx [hunk 1]
```diff
@@ -223,6 +227,136 @@ const MainNavBar = () => {
...
+  const hourMs = 60 * minuteMs;
+  const dayMs = 24 * hourMs;
+
+  if (ageMs < minuteMs) {
+    return "Updated just now";
+  }
+
+  if (ageMs < hourMs) {
+    return `Updated ${Math.floor(ageMs / minuteMs)}m ago`;
+  }
```
Comment (edoakes) [inline python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx]: In my experience the common case is that users expect the dashboard to refresh on the order of ~seconds.

Could we format the freshness label in 5s increments when under 1min? So it would say "just now" if under 5s, then nearest 5s, then `1min ago` once we hit 1 minute.

File: python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx [hunk 2]
```diff
@@ -223,6 +227,136 @@ const MainNavBar = () => {
...
+        DASHBOARD_DATA_LOADED_EVENT,
+        handleDashboardDataLoaded,
+      );
+    };
+  }, []);
+
+  React.useEffect(() => {
+    const interval = window.setInterval(() => {
+      setNow(Date.now());
+    }, 30000);
```
Comment (edoakes) [inline python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx]: I am not too familiar with react/typescript. Does this mean we only evaluate the label every 30s? Wouldn't that mean it would say "just now" for up to 30s?

**Review by edoakes — COMMENTED** [review]
Thanks for the contribution. Just one primary comment about the frequency intervals.

Do you have a sense of how much overhead this operation has? My expectation is that evaluating this every few seconds would not add notable overhead.

**Comment by edoakes** [comment]
I've also kicked off the pre-merge CI tests: https://buildkite.com/ray-project/premerge/builds/67089

File: python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx (hunk 1 above)
Comment (tdat1465) [reply python/ray/dashboard/client/src/pages/layout/MainNavLayout.tsx]: Thanks for the review!

That makes sense. I’ll update the freshness label to refresh every 5 seconds and format values under 1 minute in 5-second increments, using “Updated just now” only for values under 5 seconds.

**Comment by tdat1465** [comment]
Hi @edoakes! Thanks for the review! I updated the label to refresh every 5s and changed the sub-minute formatting to show “just now” under 5s, then rounded 5s increments until it reaches 1m.

I also added test coverage for the new boundaries and pushed the update to the PR branch.

I also re-ran the related frontend checks. Please take another look when convenient. Thanks!

**Review by edoakes — APPROVED** [review]
Thank you for the contribution
